### ISSUE TITLE
Thrash out team stance on usage of Streams in JUNG

### ISSUE THREAD

**Issue by jbduncan** [jrtom/jung#179]
In response to #148, I'm opening this issue to start a discussion on whether we should use Streams (the `java.util.stream.Stream` kind) in JUNG, and if so where or when we should use them specifically.

So, to kick off the discussion, my opinion is to use Streams whenever "reasonably" possible, e.g. for brevity or readability reasons (I know, very precise definition!), and to only use imperative alternatives if performance metrics dictate so. 

However, having said this, it's likely that my opinion will change when I finally get my copy of Effective Java 3rd Edition, which I understand has a new section on the judicious use of Streams (compared to 2nd Edition). So just to be prepared if (when?) my opinion unexpectedly changes. :)

**Comment by jrtom** [comment]
I don't have a specific guideline to propose here; Streams can either increase or decrease readability depending on how they're used and the context.

I also don't really want to get into performance-based arguments at this time, except those that are related to asymptotic differences (i.e., I'm totally willing to consider replacing an O(n lg n) algorithm with an O(n) algorithm, but I'm less interested in constant-time differences just now).  

I would like to focus my time, at least, on getting the APIs and the documentation into shape for 3.0, rather than review (or make) changes relating to Streams.  As long as there are no API implications, we can make such changes in minor releases after 3.0 is out.

Bottom line: I don't think that we need (or are likely to get) a team stance on the care and feeding of Streams at this time.  I am neither opposed to their use nor greatly inclined to encourage it.

**Comment by jbduncan** [comment]
Okey dokey, I have no objections to this. :)
